include/SampleLoader.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * SampleLoader.h - Load audio and waveform files
+ *
+ * Copyright (c) 2023 saker <sakertooth@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_GUI_SAMPLE_LOADER_H
+#define LMMS_GUI_SAMPLE_LOADER_H
+
+#include <QString>
+#include <memory>
+
+#include "SampleBuffer.h"
+#include "lmms_export.h"
+
+namespace lmms::gui {
+class LMMS_EXPORT SampleLoader
+{
+public:
+	static QString openAudioFile(const QString& previousFile = "");
+	static QString openWaveformFile(const QString& previousFile = "");
+	static std::unique_ptr<SampleBuffer> createBufferFromFile(const QString& filePath);
+	static std::unique_ptr<SampleBuffer> createBufferFromBase64(
```

Comment:
`SampleBuffer`s are only ever stored in shared pointers, not unique ones. Also, it's more efficient to create a shared pointer in the first place than to create a unique pointer and convert it into a shared one. As such, I would be tempted to have these return `std::shared_ptr<const SampleBuffer>` instead. (Similarly for the two (!?) recording classes.)